improver/calibration/rainforest_calibration.py

```diff
@@ -419,13 +465,15 @@ def _calculate_error_probabilities(
         """
         error_probability_cube = self._prepare_error_probability_cube(forecast_cube)
 
-        input_dataset = self._prepare_features_dataframe(feature_cubes)
+        input_dataset, feature_variables = self._prepare_features_array(feature_cubes)
 
-        for threshold_index, model in enumerate(self.tree_models):
-            prediction = model.predict(input_dataset)
-            error_probability_cube.data[threshold_index, ...] = np.reshape(
-                prediction, forecast_cube.data.shape
-            )
+        forecast_variable = forecast_cube.name()
+        precipitation_ind = feature_variables.index(forecast_variable)
+        precip_forecast = input_dataset[:, precipitation_ind]
```

Comment:
In fact, looking over the code again I think it would be better to get the forecast dataset from the forecast_cube rather than from the feature_cubes. This would avoid the need to look up the index, and would ensure that code still works in the case where the forecast variable is not a feature.

Reply:
Good suggestion, thanks.